### PR TITLE
RDM-7067-Automate Document AM Scenarios for F-044: Submit event creat…

### PR DESCRIPTION
**Pull request by shrivsksavit**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-438

### Change description ###

Automate Document AM Scenarios for F-044: Submit event creation as Case worker

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/aat/resources/features/F-044/F-044-Prerequisite.td.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "headers": {
       "Content-Encoding" : "gzip",
       "Content-Type" : "application/json;charset=UTF-8",
-      "Content-Length": "[[ANY]]"
+      "Content-Length": "[[ANY]]",
+      "Transfer-Encoding" : "chunked"
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/F-044-Prerequisite.td.json]: removed done

File: src/aat/resources/features/F-044/F-044.feature [hunk 2]
```diff
@@ -95,6 +95,26 @@ Feature: F-044: Submit event creation as Case worker
     And the response [contains an HTTP 404 'Not Found']
     And the response has all the details as expected
 
+  @S-579 # DM store related scenario
+  Scenario: must update successfully all and only the respective fields with update access for a Case Worker and a Solicitor
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/F-044.feature]: done

File: src/aat/resources/features/F-044/F-044.feature [hunk 3]
```diff
@@ -95,6 +95,26 @@ Feature: F-044: Submit event creation as Case worker
...
+    Given a user with [an active Case Worker profile in CCD with a specific variation of ACLs on a case type]
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/F-044.feature]: Done

File: src/aat/resources/features/F-044/S-579.td.json [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+          "value": []
+        }
+      },
+      "after_submit_callback_response": null,
+      "callback_response_status_code": null,
+      "callback_response_status": null,
+      "delete_draft_response_status_code": null,
+      "delete_draft_response_status": null,
+      "security_classifications": {
+        "DocumentField4": "PUBLIC",
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579.td.json]: done

File: src/aat/resources/features/F-044/S-579.td.json [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        ],
+        "YesOrNoField": "Yes",
+        "EmailField": "testemail@gmail.com",
+        "TextField": "sometextfield",
+        "DateField": "2222-01-02",
+        "TextAreaField": "textarea",
+        "CollectionField": []
+      },
+      "data_classification": {
+        "DocumentField4": "PUBLIC",
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579.td.json]: done

File: src/aat/resources/features/F-044/S-579.td.json [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    "body": {
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 1,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579.td.json]: done

File: src/aat/resources/features/F-044/S-579-Update_Case_For_Case_Worker_Data_Base.td.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
+{
+  "_guid_": "S-579_Update_Case_For_Case_Worker_Data_Base",
```
[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579-Update_Case_For_Case_Worker_Data_Base.td.json]: done

[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579-Update_Case_For_Case_Worker_Data_Base.td.json]: done

[the remark this answers is not in the document]
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579-Update_Case_For_Case_Worker_Data_Base.td.json]: done

**Comment by shrivsksavit** [comment]
On Jenkins my story S-579 - PASS PFA for more details 
https://build.platform.hmcts.net/job/HMCTS_CDM/job/ccd-data-store-api/job/PR-732/26/artifact/target/cucumber/cucumber-html-reports/report-feature_11_1278901126.html

File: src/aat/resources/features/F-044/S-579_Get_Case_For_Case_Worker_Data_Base.td.json [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": "[[ANY]]",
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
+        "MoneyGBPField": "111100",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Get_Case_For_Case_Worker_Data_Base.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 1,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
+        "DocumentField4": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: Can use inheritance for case_data.

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+        ],
+        "YesOrNoField": "Yes",
+        "EmailField": "testemail@gmail.com",
+        "TextField": "sometextfield",
+        "DateField": "2222-01-02",
+        "TextAreaField": "textarea",
+        "CollectionField": []
+      },
+      "data_classification": {
+        "DocumentField4": "PUBLIC",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+          "value": [],
+          "classification": "PUBLIC"
+        }
+      },
+      "after_submit_callback_response": null,
+      "callback_response_status_code": null,
+      "callback_response_status": null,
+      "delete_draft_response_status_code": null,
+      "delete_draft_response_status": null,
+      "security_classifications": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+          "document_filename": "Screenshot 2019-11-13 at 11.25.45 am.png",
+          "document_binary_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/76976634-374f-472b-a819-a3a938b4cc83/binary"
+        },
+        "DocumentField1": {
+          "document_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/e636eef3-839a-48cd-96f0-dad43e66f5c4",
+          "document_filename": "Screenshot 2019-11-13 at 3.24.26 pm.png",
+          "document_binary_url": "http://dm-store-aat.service.core-compute-aat.internal/documents/e636eef3-839a-48cd-96f0-dad43e66f5c4/binary"
+        }
+      },
+      "data_classification": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+        "EmailField": "PUBLIC",
+        "TextField": "PUBLIC",
+        "DateField": "PUBLIC",
+        "TextAreaField": "PUBLIC",
+        "CollectionField": {
+          "value": [],
+          "classification": "PUBLIC"
+        }
+      },
+      "security_classifications": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+    "body": {
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 2,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+          "OPTION3"
+        ],
+        "YesOrNoField": "Yes",
+        "EmailField": "testemail@gmail.com",
+        "TextField": "sometextfield",
+        "DateField": "2222-01-02",
+        "TextAreaField": "textarea",
+        "CollectionField": []
+      },
+      "data_classification": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+          "classification": "PUBLIC",
+          "value": []
+        }
+      },
+      "after_submit_callback_response": null,
+      "callback_response_status_code": null,
+      "callback_response_status": null,
+      "delete_draft_response_status_code": null,
+      "delete_draft_response_status": null,
+      "security_classifications": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json]: Can use inheritance.

File: src/aat/resources/features/F-044/S-579_Get_Case_For_Case_Worker_Data_Base.td.json (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Get_Case_For_Case_Worker_Data_Base.td.json]: we had tried but not possible at this moment

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json (hunk 9 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: no plan for reuse at the moment

File: src/aat/resources/features/F-044/S-579_Get_Case_For_Case_Worker_Data_Base.td.json (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Get_Case_For_Case_Worker_Data_Base.td.json]: No, we can't use Standard_Full_Case_Data as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json (hunk 9 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: No, we can't use Standard_Full_Case_Data as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json (hunk 10 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: No, we can't use Standard_Full_Case_Data as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json (hunk 9 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: No, we can't use as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json (hunk 11 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_1.td.json]: No, we can't use as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json (hunk 12 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json]: No, we can't use as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json (hunk 13 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Read_By_Solicitor_3.td.json]: No, we can't use as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json (hunk 14 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json]: No, we can't use as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json (hunk 15 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json]: No, we can't use as the data is different

File: src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json (hunk 16 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-044/S-579_Later_Case_Update_By_Solicitor_3.td.json]: No, we can't use as the data is different

**Pull request merged**